File: 프로그래머스/lv0/120866. 안전지대/안전지대.java
class Solution {
    public void danger(int[][] copy, int N, int X, int Y){
        for(int x=X-1; x<=X+1; x++){
            if(x<0 || x>=N){
                continue;
            }else{
                for(int y=Y-1; y<=Y+1; y++){
                    if(y<0 || y>=N){
                        continue;
                    }else{
                        copy[x][y] = 1;
                    }
                }
            }
        }
    }
    public int solution(int[][] board) {
        int answer = 0, n = board.length;
        int[][] copy = new int[n][n];
        
        for(int x=0; x<n; x++){
            for(int y=0; y<n; y++){
                if(board[x][y]==1){
                    danger(copy, n, x, y);
                }
            }
        }
        for(int x=0; x<n; x++){
            for(int y=0; y<n; y++){
                if(copy[x][y]==0){
                    answer++;
                }
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/lv0/120866. 안전지대/안전지대.js
function solution(board) {
    var answer = 0;
    var n = board.length;
    for(var i=0; i<n; i++){
        for(var j=0; j<n; j++){
            if(board[i][j] === 0){
                for(var ii=i-1; ii<i+2; ii++){
                    if(ii<0 || ii>=n) continue;
                    for(var jj=j-1; jj<j+2; jj++){
                        if(jj<0 || jj>=n) continue;
                        if(board[ii][jj] === 1) {
                            board[i][j] = 2;
                        }
                    }
                }
                if(board[i][j]===0) answer++;
            }
        }
    }
    return answer;
}